### PR TITLE
Fix non-linux builds

### PR DESCRIPTION
**Pull request by klauspost**
Adding build tags for linux and added stubs for other operating systems. This should ensure that code using it will always compile, but only work on linux.

I am unsure if FreeBSD and friends are expected to work, but for now it is "linux or nothing".

### CODE REVIEW

**Comment by anatol** [comment]
Hi @klauspost thank you for your contribution (+ other useful libraries you released)! multi-platform support is the top requested feature and it is definitely that need to be done.

In fact I started working on it as well, see `multi-platform` branch that you have in your forked repo. It [does pretty much the same as what you did](https://github.com/klauspost/smart.go/commit/a74390020baa28a55e08177f0ee4dc48dd3348b1) but package os specific in one file e.g. `os_linux.go`.

+ I also work on MacOSX support, see it here https://github.com/anatol/smart.go/blob/multi-platform/os_darwin.go

If you need this os separate to unblock your work then I can merge your change and rebase my changed on top of your. Or another option you can wait till I complete my work (will take a week or two). Let me know what do you think.

File: examples/nvme_linux_test.go [hunk 1]
```diff
@@ -3,10 +3,11 @@ package test
 import (
 	"bytes"
 	"fmt"
-	"github.com/anatol/smart.go"
-	"github.com/stretchr/testify/require"
 	"os/exec"
 	"testing"
+
+	"github.com/anatol/smart.go"
```
Comment (anatol) [inline examples/nvme_linux_test.go:9]: In my other project I was using such grouping. But Intellij IDEA keeps merging it back into one set when I do file modifications. I am tired to fight with Intellij and decided to give a try for "one import group" idea here.

In addition neither `go fmt` nor `revive` seems care about imports grouping. Is there any code style that says third-party imports need to be a separate group?

**Comment by anatol** [comment]
And the final goal is to support as many platforms as possible. I see that `smartmontools` has a broad OS coverage. I would love to see the same for smart.go.

File: examples/nvme_linux_test.go (hunk 1 above)
Comment (klauspost) [reply examples/nvme_linux_test.go:9]: Hmmm.. Using Goland myself, but I set up `goimports`. You should be able to change it.

**Comment by klauspost** [comment]
> If you need this os separate to unblock your work then I can merge your change and rebase my changed on top of your. Or another option you can wait till I complete my work (will take a week or two). Let me know what do you think.

I can wait a week or two. If you open a PR we can track there, and this can be closed.

As I propose, ideally the exported functions should be the same. That would mean the package can be used without build tags which makes it much easier to use. I don't know if the structs can be presented platform independent manner.

**Comment by anatol** [comment]
After thinking for a while I've decided to go with your PR instead. It has a platform-specific files for each device type that might be a cleaner structure.

Thank you for your contribution @klauspost!

**Comment by anatol** [comment]
https://github.com/anatol/smart.go/commit/4d973e3143b78ddf4b54d09a14261c27f0fc71ca

File: examples/nvme_linux_test.go (hunk 1 above)
Comment (anatol) [reply examples/nvme_linux_test.go:9]: Thanks! I knew there should be a way to do it. I am surprised that I have not found it earlier.

**Pull request closed without merging**